test/commands/pull.js

```diff
@@ -0,0 +1,144 @@
+                    name: 'requirements.txt',
+                    format: 'TEXT',
+                    content: '# Add your dependencies here.\n# See https://pip.pypa.io/en/latest/reference/requirements-file-format/\n# for how to format them\napify ~= 1.0.0\n',
+                },
+            ],
+        },
+    ],
+};
+
+const TEST_ACTOR_GITHUB_GIST = {
+    name: ACTOR_NAME,
+    isPublic: false,
+    versions: [
+        {
+            versionNumber: '0.0',
+            sourceType: 'GITHUB_GIST',
+            buildTag: 'latest',
+            gitHubGistUrl: 'https://gist.github.com/DennisKallerhoff/32f1efd686e11f6a05f1af87bddb1f1a',
+        },
+    ],
+};
+
+const TEST_ACTOR_GIT_REPO = {
+    name: ACTOR_NAME,
+    isPublic: false,
+    versions: [
+        {
+            versionNumber: '0.0',
+            sourceType: 'GIT_REPO',
+            buildTag: 'latest',
+            gitRepoUrl: 'https://github.com/HonzaTuron/baidu-scraper#multipage',
+        },
+    ],
+};
+
+describe('apify pull', () => {
+    beforeEach(() => {
+        sinon.spy(console, 'log');
+    });
+
+    it('should work with actor SOURCE_FILES', async () => {
+        const testActor = await testUserClient.actors().create(TEST_ACTOR_SOURCE_FILES);
+        const testActorClient = testUserClient.actor(testActor.id);
+        const actorFromServer = await testActorClient.get();
+
+        await command.run(['pull', testActor.id]);
+
+        const actorJson = loadJson.sync(path.join(testActor.name, LOCAL_CONFIG_PATH));
+
+        expect(actorJson.name).to.be.eql(actorFromServer.name);
+    });
+
+    it('should work with GITHUB_GIST', async () => {
+        const testActor = await testUserClient.actors().create(TEST_ACTOR_GITHUB_GIST);
+
+        await command.run(['pull', testActor.id]);
+
+        const actorPackageJson = loadJson.sync(path.join(testActor.name, 'package.json'));
+
+        expect(actorPackageJson.name).to.be.eql('act-in-gist');
+    });
+
+    it('should work with GIT_REPO', async () => {
+        const testActor = await testUserClient.actors().create(TEST_ACTOR_GIT_REPO);
+
+        await command.run(['pull', testActor.id]);
+
+        const actorJson = loadJson.sync(path.join(testActor.name, DEPRECATED_LOCAL_CONFIG_NAME));
+
+        expect(actorJson.name).to.be.eql('baidu-scraper');
+    });
+
+    it('should work without actorId', async () => {
+        const testActor = await testUserClient.actors().create(TEST_ACTOR_SOURCE_FILES);
+
+        const contentBeforeEdit = JSON.parse(TEST_ACTOR_SOURCE_FILES.versions[0].sourceFiles[2].content);
+        contentBeforeEdit.id = testActor.id;
+        TEST_ACTOR_SOURCE_FILES.versions[0].sourceFiles[2].content = contentBeforeEdit;
+
+        await writeJsonFile(LOCAL_CONFIG_PATH, TEST_ACTOR_SOURCE_FILES.versions[0].sourceFiles[2].content);
+
+        await command.run(['pull']);
+
+        const actorJson = loadJson.sync(LOCAL_CONFIG_PATH);
+
+        expect(actorJson.id).to.be.eql(testActor.id);
```

Comment:
This isn't a very good test - it would pass even if the pull did not happen at all, since the `id` is there in the file from before. It would be better to check some other file.